numpy/ma/core.py

```diff
@@ -4286,22 +4297,45 @@ def count(self, axis=None):
         array([3, 0])
 
         """
+        kwargs = {} if keepdims is np._NoValue else {'keepdims': keepdims}
+
         m = self._mask
-        s = self.shape
+        # special case for matrices (we assume no other subclasses modify
+        # their dimensions)
+        if isinstance(self.data, np.matrix):
+            if m is nomask:
+                m = np.zeros(self.shape, dtype=np.bool_)
+            m = m.view(type(self.data))
+
         if m is nomask:
-            if axis is None:
+            # compare to _count_reduce_items in _methods.py
+
+            if self.shape is ():
+                return 1
```

Comment:
Why do I only see things on fourth read? I see that this is the same as `_count_reduce_items`, but isn't it simply wrong?  Shouldn't `count` be 0 for an empty array? That is what is being returned for `axis=None` in current master (while any given value of `axis` raises `IndexError`).